yt_dlp/extractor/servus.py

```diff
@@ -60,85 +84,54 @@ class ServusIE(InfoExtractor):
-            elif extension == 'mp4' or ext == 'mp4':
-                formats.append({
-                    'url': format_url,
-                    'format_id': type_,
-                    'width': int_or_none(resource.get('width')),
-                    'height': int_or_none(resource.get('height')),
-                })
+            'https://api-player.redbull.com/stv/servus-tv?timeZone=Europe/Berlin',
+            video_id, 'Downloading video JSON', query={'videoId': video_id})
+        if 'videoUrl' not in video:
+            self._report_errors(video)
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            video['videoUrl'], video_id, 'mp4', m3u8_id='hls')
 
-        attrs = {}
-        for attribute in video['attributes']:
-            if not isinstance(attribute, dict):
-                continue
-            key = attribute.get('fieldKey')
-            value = attribute.get('fieldValue')
-            if not key or not value:
-                continue
-            attrs[key] = value
-
-        title = attrs.get('title_stv') or video_id
-        alt_title = attrs.get('title')
-        description = attrs.get('long_description') or attrs.get('short_description')
-        series = attrs.get('label')
-        season = attrs.get('season')
-        episode = attrs.get('chapter')
-        duration = float_or_none(attrs.get('duration'), scale=1000)
+        season = video.get('season')
         season_number = int_or_none(self._search_regex(
             r'Season (\d+)', season or '', 'season number', default=None))
+        episode = video.get('chapter')
         episode_number = int_or_none(self._search_regex(
             r'Episode (\d+)', episode or '', 'episode number', default=None))
 
         return {
             'id': video_id,
-            'title': title,
-            'alt_title': alt_title,
-            'description': description,
-            'thumbnail': thumbnail,
-            'duration': duration,
-            'timestamp': unified_timestamp(video.get('lastPublished')),
-            'series': series,
+            'title': video.get('title'),
+            'description': self._get_description(video_id) or video.get('description'),
+            'thumbnail': video.get('poster'),
+            'duration': float_or_none(video.get('duration')),
+            'timestamp': unified_timestamp(video.get('currentSunrise')),
+            'series': video.get('label'),
             'season': season,
             'season_number': season_number,
             'episode': episode,
             'episode_number': episode_number,
             'formats': formats,
+            'subtitles': subtitles,
         }
+
+    def _get_description(self, video_id):
+        info = self._download_json(
+            f'https://backend.servustv.com/wp-json/rbmh/v2/media_asset/aa_id/{video_id}?fieldset=page',
+            video_id, fatal=False)
+        return traverse_obj(info, 'stv_long_description', 'stv_short_description')
```

Comment:
single use function. This could go directly in _real_extract imo, but I won't insist